docs/client_advanced.rst

```diff
@@ -52,19 +52,52 @@ directly as shown above, but it is more convenient to use special keyword
 
     await session.post(url, json={'example': 'text'})
 
-For *text/plain* ::
+For ``text/plain``::
 
     await session.post(url, data='Привет, Мир!')
 
+Authentication
+--------------
+
+Instead of setting the ``Authorization`` header directly,
+:class:`ClientSession` and individual request methods provide an ``auth``
+argument. An instance of :class:`BasicAuth` can be passed in like this::
+
+    auth = BasicAuth(login="...", password="...")
+    async with ClientSession(auth=auth) as session:
+        ...
+
+For other authentication flows, the ``Authorization`` header can be set
+directly::
+
+    headers = {"Authorization": "Bearer eyJh...0M30"}
+    async with ClientSession(headers=headers) as session:
+        ...
+
+The authentication header for a session may be updated as and when required.
+For example::
+
+    session.headers["Authorization"] = "Bearer eyJh...1OH0"
+
+Note that a *copy* of the headers dictionary is set as an attribute when
+creating an :class:`ClientSession` instance (as a :class:`CIMultiDict` object).
+Updating the original dictionary does not have any effect. Instead, the
+:attr:`ClientSession.headers` property should be updated like a standard
+dictionary.
+
+In cases where the authentication header value expires periodically, an
+:mod:`asyncio` task may be used to update the session's default headers in the
+background.
+
 .. note::
 
-   ``Authorization`` header will be removed if you get redirected
-   to a different host or protocol, except the case when  ``HTTP -> HTTPS``
+   The ``Authorization`` header will be removed if you get redirected
+   to a different host or protocol, except the case when  HTTP to HTTPS
    redirect is performed on the same host.
 
 .. versionchanged:: 4.0
 
-   Started keeping the ``Authorization`` header during ``HTTP -> HTTPS``
+   Started keeping the ``Authorization`` header during HTTP to HTTPS
```

Comment:
I'd be inclined to revert these. They make the English slightly awkward and would need to become a little more wordy to make a natural sounding sentence.

Reply:
Understood. I have put the arrow back, but still proposing to drop the double back tick literal value syntax if that's ok. I found that a little confusing since we're not referencing some system identifiers or variable names etc.